compiler/src/dmd/expression.d

```diff
@@ -2126,6 +2132,16 @@ extern (C++) final class AssocArrayLiteralExp : Expression
  */
 extern (C++) final class StructLiteralExp : Expression
 {
+    struct BitFields
+    {
+        bool useStaticInit;     /// if this is true, use the StructDeclaration's init symbol
+        bool isOriginal = false; /// used when moving instances to indicate `this is this.origin`
+        OwnedBy ownedByCtfe = OwnedBy.code;
```

Comment:
any reason not to put these into `Expression`'s BitFields?

Reply:
That would make them accessible to non-StructLiteral Expression nodes, and since there's a 4 byte gap between `Loc` and `sd` because of alignment, it wouldn't make a difference.